packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -12,11 +13,27 @@ import 'package:flutter/widgets.dart';
 import 'colors.dart';
 import 'theme.dart';
 import 'typography.dart';
+import 'constants.dart';
 
+const Duration _kDialAnimateDuration = const Duration(milliseconds: 200);
+const double _kTwoPi = 2 * math.PI;
+const int _kHoursPerDay = 24;
+const int _kHoursPerPeriod = 12;
+const int _kMinutesPerHour = 60;
+
+enum DayPeriod {
+  am,
+  pm,
+}
+
+/// A value representing a time during the day
 class TimeOfDay {
```

Comment:
Yeah, we should figure out a way to use the DateFormat class.  The issue is that it works with a DateTime, which is a specific date and time (e.g., of a particular year and day).  The picker works on an abstract time of day that isn't tied to a specific day.  We'd have to find a prototype day that has a normal set of times.  Also every day will work, but you need to avoid days that have daylight savings or leap second magic.